cloud/amazon/lambda.py

```diff
@@ -0,0 +1,394 @@
+    HAS_HASHLIB = False
+
+
+def get_current_function(connection, function_name):
+    try:
+        return connection.get_function(FunctionName=function_name)
+    except botocore.exceptions.ClientError as e:
+        return False
+
+
+def sha256sum(filename):
+    hasher = hashlib.sha256()
+    with open(filename, 'rb') as f:
+        hasher.update(f.read())
+
+    code_hash = hasher.digest()
+    code_b64 = base64.b64encode(code_hash)
+    hex_digest = code_b64.decode('utf-8')
+
+    return hex_digest
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+        name=dict(type='str', required=True),
+        state=dict(type='str', default='present', choices=['present','absent']),
+        runtime=dict(type='str', choices=['nodejs','java8','python2.7'], default=None),
+        role_arn=dict(type='str', default=None),
+        handler=dict(type='str', default=None),
+        path=dict(type='str', default=None, aliases=['src']),
+        s3_bucket=dict(type='str', default=None),
+        s3_key=dict(type='str', default=None),
+        s3_object_version=dict(type='str', default=None),
+        description=dict(type='str', default=''),
+        timeout=dict(type='int', default=3),
+        memory_size=dict(type='int', default=128),
+        )
+    )
+
+    module = AnsibleModule(argument_spec=argument_spec,
+                           supports_check_mode=True,
+                           mutually_exclusive=[['path', 's3_key'],
+                                               ['path', 's3_bucket'],
+                                               ['path', 's3_object_version']])
+
+    name = module.params.get('name')
+    state = module.params.get('state').lower()
+    runtime = module.params.get('runtime')
+    role_arn = module.params.get('role_arn')
+    handler = module.params.get('handler')
+    s3_bucket = module.params.get('s3_bucket')
+    s3_key = module.params.get('s3_key')
+    s3_object_version = module.params.get('s3_object_version')
+    path = module.params.get('path')
+    description = module.params.get('description')
+    timeout = module.params.get('timeout')
+    memory_size = module.params.get('memory_size')
+
+    check_mode = module.check_mode
+    changed = False
+
+    if not HAS_BOTO:
+        module.fail_json(msg='Python module "boto" is missing, please install it')
+
+    if not HAS_BOTO3:
+        module.fail_json(msg='Python module "boto3" is missing, please install it')
+
+    if not HAS_BINASCII:
+        module.fail_json(msg='Python module "binascii" is missing, please install it')
+
+    if not HAS_HASHLIB:
+        module.fail_json(msg='Python module "hashlib" is missing, please install it')
+
+    if not HAS_BASE64:
+        module.fail_json(msg='Python module "base64" is missing, please install it')
+
+
+    try:
+        region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
+        client = boto3_conn(module, conn_type='client', resource='lambda',
```

Comment:
You should check that region is set here before connection.  See other modules for examples.